.script/tests/detectionTemplateSchemaValidation/DetectionTemplateSchemaValidationTests.cs

```diff
@@ -10,19 +12,50 @@
 
 namespace Kqlvalidations.Tests
 {
-    public class DetectionTemplateStructureValidationTests
+    public class DetectionTemplateSchemaValidationTests
     {
         private static readonly string DetectionPath = DetectionsYamlFilesTestData.GetDetectionPath();
 
         [Theory]
         [ClassData(typeof(DetectionsYamlFilesTestData))]
         public void Validate_DetectionTemplates_HaveValidTemplateStructure(string detectionsYamlFileName)
+        {
+            Action<ScheduledTemplateInternalModel> validateConnectorSchema = (ScheduledTemplateInternalModel templateObject) => {
+                var validationContext = new ValidationContext(templateObject);
+                Validator.ValidateObject(templateObject, validationContext, true);
+            };
+
+            ValidateDetectionTemplate(detectionsYamlFileName, validateConnectorSchema);
+        }
+
+        [Theory]
+        [ClassData(typeof(DetectionsYamlFilesTestData))]
+        public void Validate_DetectionTemplates_HaveValidConnectorsIds(string detectionsYamlFileName)
+        {
+            Action<ScheduledTemplateInternalModel> validateConnectorsIds = (ScheduledTemplateInternalModel templateObject) => {
```

Comment:
IMO you should not reuse the code for ValidTemplateStructure - meaning not using TOObject<ScheduledTemplateInternalModel > becuse this is not what we want to check here
you can just do:
 var deserializer = new DeserializerBuilder().Build();
            var res = deserializer.Deserialize<dynamic>(yaml);
            string queryStr = res["RequiredDataConnectors"];